skyportal/models/photometry.py

```diff
@@ -248,3 +263,720 @@ def snr(self):
+
+        Returns
+        -------
+        float array
+            Array of fluxes in units of micro Jansky.
+        """
+
+        fluxes = PhotometricSeries.mag2flux(mags)
+        return fluxes * magerr * np.log(10) / 2.5
+
+    def calc_flux_mag(self):
+        """
+        Use the available fluxes to calculate the magnitudes,
+        or the available magnitudes to calculate the fluxes.
+        Also fills in the mjds field, and the errors if
+        they are included in the dataset.
+        """
+
+        if 'fluxes' in self._data:
+            self._fluxes = self._data['fluxes']
+            self._mags = self.flux2mag(self._fluxes)
+        elif 'mags' in self._data:
+            self._mags = self._data['mags']
+            self._fluxes = self.mag2flux(self._mags)
+        else:
+            raise KeyError('Cannot find "fluxes" or "mags" in photometric data')
+
+        if 'fluxerr' in self._data:
+            self._fluxerr = self._data['fluxerr']
+            self._magerr = self.fluxerr2magerr(self.fluxes, self._fluxerr)
+        elif 'magerr' in self._data:
+            self._magerr = self._data['magerr']
+            self._fluxerr = self.magerr2fluxerr(self._mags, self._magerr)
+
+    def calculate_stats(self):
+        """
+        Calculate some summary statistics to be saved in the DB.
+
+        Will fill the JSONB columns of medians, minima and maxima
+        and standard deviations for each column in the input data,
+        including fluxes, but also any other auxiliary data
+        e.g., backgrounds, raw counts, sensor X/Y position.
+
+        Will also find the start/end/midpoint of the
+        series MJD, the mean magnitude, the magnitude RMS,
+        and the robust RMS using sigma-clipping.
+        """
+
+        # get the min/max/media for each column of data
+        self.medians = {}
+        self.minima = {}
+        self.maxima = {}
+        self.stds = {}
+        for key in self._data:
+            self.medians[key] = float(self._data.median('mjds'))
+            self.minima[key] = float(self._data.min('mjds'))
+            self.maxima[key] = float(self._data.max('mjds'))
+            self.stds[key] = float(self._data.std('mjds'))
+
+        self.mjd_first = self.mjds[0]
+        self.mjd_last = self.mjds[-1]
+        self.mjd_mid = (self.mjd_start + self.mjd_end) / 2
+        detection_indices = np.where(self.snr > PHOT_DETECTION_THRESHOLD)[0]
+        self.mjd_last_detected = self.mjd[detection_indices[-1]]
+        self.num_exp = len(self.mjds)
+
+        # calculate the mean mag, rms and robust median/rms
+        self.mean_mag = np.nanmean(self.mags)
+        self.rms_mag = np.nanstd(self.mags)
+        (self.robust_mag, self.robust_rms) = self.sigma_clipping(self.mags)
+
+        # if RA, Dec or exposure time are given in the auxiliary data:
+        for key in ['RA', 'ra']:
+            if key in self._data:
+                self.ra = self._data[key].median()
```

Comment:
should be nanmedian in case this is missing in some entries

Reply:
The xarray median skips NaNs by default.